### ISSUE TITLE
4.3.9 Backported issues

### ISSUE THREAD

**Issue by fmbenhassine** [spring-projects/spring-batch#4340]
* #4024
* #4101
* #4246
* #4402
* #4381
* #4092

**Comment by hjlee83** [comment]
Hi!

How about this?

```java
private static final String FIND_LAST_JOB_INSTANCE_BY_JOB_NAME = """
			SELECT MAX(JOB_INSTANCE_ID), JOB_NAME
			FROM %PREFIX%JOB_INSTANCE
			WHERE JOB_NAME = ?
                        GROUP BY JOB_NAME""";
```

**Comment by fmbenhassine** [comment]
@hjlee83 I am not sure I understand your question. Can you elaborate please? Are you referring to the formatting of the query using Java's text blocks or the port of the fix itself? The fix itself was already ported to 4.3.x, see 81e619b4332c2d669a306ed3efa1b9a9dc6cd4f7.

**Comment by fmbenhassine** [comment]
Closing this, as v4.3.9 has been released: https://github.com/spring-projects/spring-batch/releases/tag/4.3.9
